src/run/runner/valgrind/measure.rs

```diff
@@ -42,6 +46,63 @@ lazy_static! {
     };
 }
 
+/// Creates the command that executes the specified command safely.
+///
+/// # Returns
+///
+/// A tuple containing:
+/// 1. The command to execute
+/// 2. The path to the wrapper shell script file
+/// 3. The path to the file containing the exit code
+fn safe_run_cmd(to_execute: &str) -> anyhow::Result<(String, TempPath, TempPath)> {
+    const WRAPPER_SCRIPT: &str = r#"
+        #!/bin/sh
+
+        safe_run()
+        {
+            (eval $1)
+            status=$?
+            echo -n "$status" > "$2"
+        }
+
+        # Args:
+        # 1. Cmd that should be executed
+        # 2. File to write the status code to
+        if [ "$#" -ne 2 ]; then
+            echo "Usage: $0 <cmd> <file>"
+            exit 1
+        fi
```

Comment:
I'm not sure we want to include usage guidelines in this since if we fail here, we will have much bigger problems. This could be some comments directly in the rust code